parsely/parselyandroid/ParselyTracker.java

```diff
@@ -258,48 +392,34 @@ private String JsonEncode(Map<String, Object> map){
 
     /*! \brief Allow Parsely to send pageview events
     *
-    *  Instantiates the callback timer responsible for flushing the events queue.
+    *  Instantiates the callback flushTimer responsible for flushing the events queue.
     *  Can be called before of after `stop`, but has no effect if used before instantiating the
     *  singleton
     */
     public void setFlushTimer(){
-        if(this.flushTimerIsActive()){
-            this.stopFlushTimer();
-        }
-        this.timer = new Timer();
-        this.timer.scheduleAtFixedRate(new TimerTask(){
-            public void run(){
-                flush();
-            }
-        }, this.flushInterval * 1000, this.flushInterval * 1000);
+        this.flushManager.start();
     }
 
-    /*! \brief Is the callback timer running
+    /*! \brief Is the callback flushTimer running
     *
-    *  @return `true` if the callback timer is currently running, `false` otherwise
+    *  @return `true` if the callback flushTimer is currently running, `false` otherwise
     */
     public boolean flushTimerIsActive(){
-        return this.timer != null;
+        return this.flushManager.isRunning();
```

Comment:
`flushTimerIsActive` is another method that could be removed from the public API in favor of having callers use `this.flushManager.isRunning()`.

Reply:
I don't want to make the manager public, but instead want to provide methods to use/access it. It creates better security for the app against someone changing the value of it and doing something unexpected.